### PR TITLE
Reader: add fast exit tracking

### PR DESCRIPTION
**Pull request by eoigal**
Fixes https://github.com/Automattic/loop/issues/264

This PR considers a fast exit as when a user exits a post in less than 25% of estimated reading time.

## Proposed Changes

- Estimated reading time is taken from the `post.minutes_to_read` added in https://github.com/Automattic/wp-calypso/pull/97529
- Adding `checkFastExit` and `trackFastExit` methods to log and track if a user exits the post before completing it, defined as exiting in less than **25%** of the estimated reading time.

## Why are these changes being made?

*

## Testing Instructions

* Go to `/read`  - add some debug to the `recordTrackForPost` function to output track event to console when hit.
* Test loading a long form post and exiting quickly - you should see console for the fastExit track event.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by spsiddarthan — APPROVED** [review]
Works well

File: client/blocks/reader-full-post/index.jsx [hunk 1]
```diff
@@ -333,6 +335,38 @@ export class FullPostView extends Component {
 		}
 	};
 
+	trackFastExit = ( post, elapsedTime, fastExitThreshold ) => {
+		recordTrackForPost( 'calypso_reader_article_fast_exit', post, {
+			context: 'full-post',
+			estimated_reading_time: post.minutes_to_read,
+			elapsed_time: elapsedTime,
```
Comment (spsiddarthan) [inline client/blocks/reader-full-post/index.jsx]: Will there be any issues because estimated_reading_time is in minutes and elasped_time is in seconds?

**Review by spsiddarthan — APPROVED** [review]
Works well,  I have asked a non blocking question about minutes and seconds both being used.

File: client/blocks/reader-full-post/index.jsx (hunk 1 above)
Comment (eoigal) [reply client/blocks/reader-full-post/index.jsx]: I'll rename the `elasped_time` to `elasped_seconds` to make it clearer

**Review by spsiddarthan — APPROVED** [review]
Thanks for the name change! Looks good to me.

**Pull request merged**